docs/upgrade/UPGRADE-unreleased.md

```diff
@@ -326,6 +326,265 @@ There you can find links to upgrade notes for other versions too.
+        - `Shopsys\FrameworkBundle\Model\Product\ProductVisibilityImmediateCronModule` has been removed
+        - `Product::recalculateVisibility` has been removed
+        - `Product::setRecalculateVisibility()` has been removed
+        - `Product::markForVisibilityRecalculation()` has been removed
+        - `Product::isMarkedForVisibilityRecalculation()` has been removed
+        - `ProductVisibilityFacade::markProductsForRecalculationAffectedByCategory()` has been replaced by `ProductVisibilityFacade::refreshProductsVisibilityByCategory()`
+        - `ProductVisibilityFacade::refreshProductsVisibilityForMarkedDelayed()` has been removed, you can use `ProductVisibilityFacade::refreshProductVisibilityById()` or `ProductVisibilityFacade::refreshProductsVisibility` instead
+        - all methods in `ProductVisibilityRepository` that has been accepting `$onlyMarkedProducts` parameter now accept `Product::id` or `null` instead
+        - `ProductFacadeTest::testEditMarkProductForVisibilityRecalculation()` has been removed
+        - [ProductVisibilityRecalculateConsumerTest](/project-base/tests/ShopBundle/Functional/Model/Product/ProductVisibilityRecalculateConsumerTest.php) has been added
+    - update all occurrences of `ProductVisibilityrepository::refreshProductVisibility()` in your `ProductVisibilityRepositoryTest` like this:
+        ```diff
+        -        $productVisibilityRepository->refreshProductsVisibility(true);
+        +        $productVisibilityRepository->refreshProductsVisibility();
+
+        ``` 
+    - update `ProductAvailabilityCalculationTest` file like this:
+        ```diff
+            use Shopsys\FrameworkBundle\Model\Product\Availability\AvailabilityFacade;
+            use Shopsys\FrameworkBundle\Model\Product\Availability\ProductAvailabilityCalculation;
+        +   use Shopsys\FrameworkBundle\Model\Product\ProductChangeMessageProducer;
+            use Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface;
+            use Shopsys\FrameworkBundle\Model\Product\ProductRepository;
+
+        ```
+        three occurrences off:
+        ```diff
+        +    $productChangeMessageProducer = $this->createMock(ProductChangeMessageProducer::class);
+        +
+             $productAvailabilityCalculation = new ProductAvailabilityCalculation(
+                 $availabilityFacadeMock,
+                 $productSellingDeniedRecalculatorMock,
+                 $productVisibilityFacadeMock,
+                 $entityManagerMock,
+        -        $productRepositoryMock
+        +        $productRepositoryMock,
+        +        $productChangeMessageProducer
+             );
+        ```
+    - update your [Dockefile](/project-base/docker/php-fpm/Dockerfile) file like this:
+        ```diff
+            opcache \
+            pgsql \
+            pdo_pgsql \
+        +   sockets \
+            zip
+        ```
+        ```diff
+            # overwrite the original entry-point from the PHP Docker image with our own
```

Comment:
```suggestion
```

It might be confused with another instruction :slightly_smiling_face:

Reply:
Done